### PR TITLE
use boost::shared_ptr instead of raw pointers for more condense code and...

### PR DESCRIPTION
**Pull request by vrabaud**
... to fix a memory leak

### CODE REVIEW

File: rviz_plugin_tutorials/src/imu_display.cpp [hunk 1]
```diff
@@ -166,15 +127,16 @@ void ImuDisplay::processMessage( const sensor_msgs::Imu::ConstPtr& msg )
...
 
-  int history_length = history_length_property_->getInt();
-
   // We are keeping a circular buffer of visual pointers.  This gets
-  // the next one, or creates and stores it if it was missing.
-  ImuVisual* visual = visuals_[ messages_received_ % history_length ];
-  if( visual == NULL )
+  // the next one, or creates and stores it if the buffer is not full
+  boost::shared_ptr<ImuVisual> visual;
+  if( visuals_.size() == visuals_.capacity() )
```
Comment (hershwg) [inline rviz_plugin_tutorials/src/imu_display.cpp]: The fact that we have to test if the number of elements equals the size limit every time we push something on pretty much means boost::circular_buffer is not an improvement over a std::deque, for this situation, no?

Comment (vrabaud) [inline rviz_plugin_tutorials/src/imu_display.cpp]: what you had before was definitely optimal (except for the resize). The goal of that rewrite was just to simplify the reading of the buffer part, to help focus on the plugin when reading the code. Whether you use that circular buffer or a list (not sure why a deque), you have to destroy a shared pointer, which is still worse than the re-assignment you had before.
BTW, there is actually a full() member function I had missed in the API. Just pushed.

**Comment by dgossow** [comment]
@hershwg What's with this? Merge? Seems like an improvement. Will test later.

**Comment by hershwg** [comment]
Right, I got distracted.  The change looks good at this point, I just needed to test it.  Which I still haven't done.

**Comment by dgossow** [comment]
I tried to test this today, but the tutorial just crashes rviz on my machine.

**Comment by vrabaud** [comment]
damn, it was running for me on Precise 64.

**Comment by dgossow** [comment]
Might as well be a bug in rviz or classloader. Will look into this later this week.

**Comment by dgossow** [comment]
ok, so the segfault probably came from somewhere else. will re-evaluate next week.

**Comment by dgossow** [comment]
Tested and works fine. Thanks Vincent!

**Pull request merged**